### ISSUE TITLE
>1440p and additional audio streams

### ISSUE THREAD

**Issue by anxdpanic** [jdf76/plugin.video.youtube#257]
>1440p and additional audio streams are VP9, and Opus codecs which are currently unsupported by inputsteam.adaptive

**Comment by anxdpanic** [comment]
We can't use anything there. We already have the dash file, our issue is playback, those instructions use web based and external app for playback. 

Using an external/alternative player like VLC 3 in Kodi and the add-on may actually work, I've never tried but everything is already in the add-on for that. The only thing that for sure won't work with external player is 'My Channel' videos.

**Comment by greystashh** [comment]
But can you say what's the problem with playback? I thought kodi is capable playing webm files

**Comment by anxdpanic** [comment]
Because it's dash, inputstream.adaptive handles the processing not core and VP9 and webm currently not supported there.

We could probably play the video stream or audio streams by themselves with core, but we can't mux them as core doesn't support that. That would require an inputstream add-on. 

In Kodi 18 we include the VP9 and webm streams in the dash file since we can and then they should work if/when supported is added to inputstream.adaptive.

**Comment by greystashh** [comment]
So the difficulty is to mux video and audio with input stream to make it playable? For mp4 this is not necessary?

**Comment by anxdpanic** [comment]
The non dash is already combined audio, and video but 720 or lower. 

The dash urls are separate audio, and video streams(see example in #53), inputstream.adaptive plays separate audio, and video streams from the dash file and currently supports mp4 so those work.
Not sure it's true muxing but I hope that clarifies.

**Comment by greystashh** [comment]
Yes. You don't know whether the method of muxing mp4 audio&video can be adopted to webm?

**Comment by anxdpanic** [comment]
afaik it can be, it just has to be written, but I don't know if or when that will happen. And I don't know any C variant so at this point I can't really contribute to that.

**Comment by greystashh** [comment]
Have you talked to peak lately again?

**Comment by anxdpanic** [comment]
I haven't, but there is an open issue with a sample and I'm sure he'll update the issue if/when he decides to undertake it, or someone else with the required skills decides to help out.

**Comment by anxdpanic** [comment]
6.4.0 enables webm-vp9, requires inputstream.adaptive >= 2.3.14
https://github.com/jdf76/plugin.video.youtube/wiki/Installation
Submitted and waiting review to the Official repository as well
